src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+    private String filePath;
+    private PrintWriter storeInFile;
+    private boolean filePresent = false;
+    private File fileCreation;
+
+    /**
+     * Constructor to create a directory or file to store the tasklist locally on the user's computer.
+     * @param filePath location of where the file should be saved
+     * @throws IOException when input or output cannot be interpreted and when the program do not have access to create a new file.
+     */
+    public Storage(String filePath) throws IOException {
+        try {
+            this.filePath = filePath;
+            String filePath2 = filePath;
+            while (filePath2.contains("/")) {
+                int slashPos = filePath2.indexOf("/");
+                File dirCreation = new File(filePath2.substring(0, slashPos));
+                if (dirCreation.mkdir()) {
+                    System.out.println("Directory has been created successfully");
+                } else {
+                    System.out.println("Directory already exist");
+                }
+                filePath2 = filePath2.substring(slashPos + 1);
+            }
+            fileCreation = new File(filePath);
+            if (fileCreation.createNewFile()) {
+                System.out.println("File created: " + fileCreation.getName());
+            } else {
+                filePresent = true;
+                System.out.println("File already exists.");
+            }
+
+        } catch (IOException e) {
+            System.out.println("Folder do not exist.");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Saves the current tasklist locally on the user's computer.
+     * @param tasklist the current working tasklist that has been created by the user.
+     * @throws FileNotFoundException if the location to store the tasklist in is not available.
+     */
+    public void saveToFile(Tasklist tasklist) throws FileNotFoundException {
+        storeInFile = new PrintWriter(fileCreation);
```

Comment:
Should storeInFile be named in a way that more explicitly and intuitively conveys its function?